modules/sync/syncs/httpsynchronization.cpp

```diff
@@ -110,14 +119,34 @@ void HttpSynchronization::start() {
     _syncThread = std::thread(
         [this](const std::string& q) {
             for (const std::string& url : _syncRepositories) {
-                const bool success = trySyncFromUrl(url + q);
-                if (success) {
-                    createSyncFile();
+                const SynchronizationState syncState = trySyncFromUrl(url + q);
+
+                // Could not get this sync repository list of files. 
+                if (syncState == SynchronizationState::ListDownloadFail) {
+                    continue;
+                }
+
+                if (syncState == SynchronizationState::Success) {
                     _state = State::Resolved;
-                    return;
+                    bool isFullySynchronized = true;
+                    createSyncFile(isFullySynchronized);
```

Comment:
Not a great fan of these variables, how about:
```suggestion
                    createSyncFile(true);
```
instead?

Alternatively:
```cpp
constexpr bool IsFullySynchronized = true;
createSyncFile(IsFullySynchronized);
```